File: ANCIEN BACK/cible/domain/stock/services/inventory_service.py
from cible.domain.stock.interface.i_inventory_repository import IInventoryRepository
from cible.domain.stock.models.inventory_basic_parts_catalog_bean import (
    InventoryBasicPartsCatalogBean,
)
from cible.domain.stock.models.inventory_ec_structuring_bean import (
    InventoryEcStructuringBean,
)
from cible.domain.stock.models.inventory_lmj_bean import InventoryLmjBean
from cible.domain.stock.models.inventory_omega_bean import InventoryOmegaBean


def get_all_inventory_basic_parts_catalog(
    repository: IInventoryRepository,
) -> list[InventoryBasicPartsCatalogBean]:
    return repository.get_all_inventory_basic_parts_catalog()


def get_all_inventory_ec_structuring(
    repository: IInventoryRepository,
) -> list[InventoryEcStructuringBean]:
    return repository.get_all_inventory_ec_structuring()


def get_all_inventory_lmj(repository: IInventoryRepository) -> list[InventoryLmjBean]:
    return repository.get_all_inventory_lmj()


def get_all_inventory_omega(
    repository: IInventoryRepository,
) -> list[InventoryOmegaBean]:
    return repository.get_all_inventory_omega()


def delete_inventory_basic_parts_catalog(
    uuid: str, repository: IInventoryRepository
) -> list[InventoryBasicPartsCatalogBean]:
    return repository.delete_inventory_basic_parts_catalog(uuid)


def create_inventory_basic_parts_catalog(
    repository: IInventoryRepository,
    inventory_basic_parts_catalog_bean: InventoryBasicPartsCatalogBean,
) -> list[InventoryBasicPartsCatalogBean]:
    return repository.create_inventory_basic_parts_catalog(
        inventory_basic_parts_catalog_bean
    )


def delete_inventory_ec_structuring(
    uuid: str, repository: IInventoryRepository
) -> list[InventoryEcStructuringBean]:
    return repository.delete_inventory_ec_structuring(uuid)


def create_inventory_ec_structuring(
    repository: IInventoryRepository,
    inventory_ec_structuring_bean: InventoryEcStructuringBean,
) -> list[InventoryEcStructuringBean]:
    return repository.create_inventory_ec_structuring(inventory_ec_structuring_bean)


def delete_inventory_lmj(
    uuid: str, repository: IInventoryRepository
) -> list[InventoryLmjBean]:
    return repository.delete_inventory_lmj(uuid)


def create_inventory_lmj(
    repository: IInventoryRepository, inventory_lmj_bean: InventoryLmjBean
) -> list[InventoryLmjBean]:
    return repository.create_inventory_lmj(inventory_lmj_bean)


def delete_inventory_omega(
    uuid: str, repository: IInventoryRepository
) -> list[InventoryOmegaBean]:
    return repository.delete_inventory_omega(uuid)


def create_inventory_omega(
    repository: IInventoryRepository, inventory_omega_bean: InventoryOmegaBean
) -> list[InventoryOmegaBean]:
    return repository.create_inventory_omega(inventory_omega_bean)


def create_inventory_basic_parts_catalog_from_excel_sheet(
    repository: IInventoryRepository,
    inventory_basic_parts_catalog_bean: InventoryBasicPartsCatalogBean,
) -> None:
    return repository.create_inventory_basic_parts_catalog_from_excel_sheet(
        inventory_basic_parts_catalog_bean
    )


def create_inventory_ec_structuring_from_excel_sheet(
    repository: IInventoryRepository,
    inventory_ec_structuring_bean: InventoryEcStructuringBean,
) -> None:
    return repository.create_inventory_ec_structuring_from_excel_sheet(
        inventory_ec_structuring_bean
    )


def create_inventory_lmj_from_excel_sheet(
    repository: IInventoryRepository, inventory_lmj_bean: InventoryLmjBean
) -> None:
    return repository.create_inventory_lmj_from_excel_sheet(inventory_lmj_bean)


def create_inventory_omega_from_excel_sheet(
    repository: IInventoryRepository, inventory_omega_bean: InventoryOmegaBean
) -> None:
    return repository.create_inventory_omega_from_excel_sheet(inventory_omega_bean)


def add_additional_comment_to_basic_parts_catalog(
    repository: IInventoryRepository, comment: str, object_uuid: str
) -> None:
    return repository.add_additional_comment_to_basic_parts_catalog(
        comment, object_uuid
    )


def add_additional_comment_to_ec_structuring(
    repository: IInventoryRepository, comment: str, object_uuid: str
) -> None:
    return repository.add_additional_comment_to_ec_structuring(comment, object_uuid)


def add_additional_comment_to_inventory_lmj(
    repository: IInventoryRepository, comment: str, object_uuid: str
) -> None:
    return repository.add_additional_comment_to_inventory_lmj(comment, object_uuid)


def add_additional_comment_to_inventory_omega(
    repository: IInventoryRepository, comment: str, object_uuid: str
) -> None:
    return repository.add_additional_comment_to_inventory_omega(comment, object_uuid)

File: ANCIEN FRONT/services/stock/inventory.service.ts
import {fetcher} from "../utils/fetch-utils.service.ts";
import {BASE_API_CIBLE_HREF} from "../../constants.ts";
import InventoryBasicPartsCatalogModel from "../../core/domain/stock/InventoryBasicPartsCatalog.ts";
import InventoryEcStructuringModel from "../../core/domain/stock/InventoryEcStructuringModel.ts";
import InventoryLmjModel from "../../core/domain/stock/InventoryLmjModel.ts";
import InventoryOmegaModel from "../../core/domain/stock/InventoryOmegaModel.ts";


export function getAllInventoryBasicPartsCatalog(): Promise<InventoryBasicPartsCatalogModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-inventory-basic-parts-catalog`, requestOptions);
}

export function getAllInventoryEcStructuring(): Promise<InventoryEcStructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-inventory-ec-structuring`, requestOptions);
}

export function getAllInventoryLmj(): Promise<InventoryLmjModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-inventory-lmj`, requestOptions);
}

export function getAllInventoryOmega(): Promise<InventoryOmegaModel[]> {
    const requestOptions: RequestInit = {
        method: 'GET',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/get-all-inventory-omega`, requestOptions);
}

export function deleteInventoryBasicPartsCatalog(uuid: string): Promise<InventoryBasicPartsCatalogModel[]> {
    const requestOptions: RequestInit = {
        method: 'DELETE',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-basic-parts-catalog/${uuid}`, requestOptions);
}

export function deleteInventoryEcStructuring(uuid: string): Promise<InventoryEcStructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'DELETE',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-ec-structuring/${uuid}`, requestOptions);
}

export function deleteInventoryLmj(uuid: string): Promise<InventoryLmjModel[]> {
    const requestOptions: RequestInit = {
        method: 'DELETE',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-lmj/${uuid}`, requestOptions);
}

export function deleteInventoryOmega(uuid: string): Promise<InventoryOmegaModel[]> {
    const requestOptions: RequestInit = {
        method: 'DELETE',
        headers: {'Content-Type': 'application/json'},
    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-omega/${uuid}`, requestOptions);
}

export function createInventoryBasicPartsCatalogModel(object : InventoryBasicPartsCatalogModel): Promise<InventoryBasicPartsCatalogModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-basic-parts-catalog`, requestOptions);
}

export function createInventoryEcStructuring(object : InventoryEcStructuringModel): Promise<InventoryEcStructuringModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-ec-structuring`, requestOptions);
}

export function createInventoryLmj(object : InventoryLmjModel): Promise<InventoryLmjModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-lmj`, requestOptions);
}

export function createInventoryOmega(object : InventoryOmegaModel): Promise<InventoryOmegaModel[]> {
    const requestOptions: RequestInit = {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(object)

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-omega`, requestOptions);
}

export function addAdditionalCommentToBasicParts(comment: string, uuid: string): Promise<null> {
    const requestOptions: RequestInit = {
        method: 'PUT',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({comment, uuid})

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-basic-parts-catalog`, requestOptions);
}

export function addAdditionalCommentToEcStructuring(comment: string, uuid: string): Promise<null> {
    const requestOptions: RequestInit = {
        method: 'PUT',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({comment, uuid})

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-ec-structuring`, requestOptions);
}
export function addAdditionalCommentToInventoryLmj(comment: string, uuid: string): Promise<null> {
    const requestOptions: RequestInit = {
        method: 'PUT',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({comment, uuid})

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-lmj`, requestOptions);
}
export function addAdditionalCommentToInventoryOmega(comment: string, uuid: string): Promise<null> {
    const requestOptions: RequestInit = {
        method: 'PUT',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({comment, uuid})

    };
    return fetcher(`${BASE_API_CIBLE_HREF}/stocks/inventory-omega`, requestOptions);
}